build.sh

```diff
@@ -7,68 +7,13 @@
 # Prevent jenkins from immediately killing the script when a step fails, allowing us to notify github:
 set +e
 
-PULLREQUEST="false"
-MERGE_FROM="master"
-while getopts 'pt:' opt; do
-    case $opt in
-        p)
-            PULLREQUEST="true"
-            ;;
-        t)
-            # hack for testing
-            MERGE_FROM="$OPTARG"
-            ;;
-        \?)
-            echo "Unknown option. supported: -p for pull request" >&2
-            exit 1
-            ;;
-    esac
-done
-shift $((OPTIND-1))
-
 REPO_ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $REPO_ROOT_DIR
 
-# GitHub notifier: reset any statuses from prior builds for this commit
-$REPO_ROOT_DIR/tools/github_update.py reset
-
 _notify_github() {
```

Comment:
For the higher level question, i'm torn.  The idea of "this is a job where it makes sense to clear status" should be in the pipeline  However, it would be better for the mechanics not to be.